server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -476,11 +476,26 @@ public long ramBytesUsed() {
     }
 
     /**
-     * Returns how much RAM would be freed up by refreshing. This is the RAM usage of the current version map. It doesn't include tombstones
-     * since they don't get cleared on refresh, nor the old version map that is being reclaimed.
+     * Returns how much RAM is used by refresh. This is the RAM usage of the current and old version maps, and the RAM usage of the
+     * archive, if any.
      */
     long ramBytesUsedForRefresh() {
-        return maps.current.ramBytesUsed.get();
+        return maps.ramBytesUsed() + archive.getMemoryBytesUsed();
+    }
+
+    /**
+     * Returns how much RAM could be reclaimed from the version map.
+     * <p>
+     * In stateful, this is the RAM usage of the current version map, and could be reclaimed by refreshing. It doesn't include tombstones
+     * since they don't get cleared on refresh, nor the old version map that is being reclaimed.
+     * <p>
+     * In stateless, this is the RAM usage of current and old version map plus the RAM usage of the parts of the archive that require a
+     * new unpromotable refresh. To reclaim all three components we need to refresh AND flush.
+     */
+    long reclaimableRamBytes() {
+        return archive == LiveVersionMapArchive.NOOP_ARCHIVE
+            ? maps.current.ramBytesUsed.get()
+            : maps.ramBytesUsed() + archive.getReclaimableMemoryBytes();
```

Comment:
In cases like this obviously the returned value is an estimate as for example we might double count the old map. I am assuming we are okay with these estimates rather than forcing synchronizations on these?